Add addon based fields, fix column formatting

harterrt opened the pull request:
Changed column names from _mon => 0_mon, removed capitalization from agg
type (e.g. geo_Mode -> geo_mode)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mozilla/telemetry-batch-view/122)
<!-- Reviewable:end -->

autolander commented:
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

vitillo commented:
Please break the commit into two separate commits, one for the name changes and the other for adding add-on fields.

harterrt commented:
Thanks, @vitillo. This is ready for another look.

mreid-moz commented on src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala:
```diff
@@ -7,6 +7,22 @@ import org.apache.spark.sql.SQLContext
+  , val app_disabled: Option[Boolean]
+  , val version: Option[String]
+  , val scope: Option[Long]
+  , val Type: Option[String] // `type` is reserved, and the spark pattern matching requires the field be present
```
you can use ``val `type`: Option[String]``

harterrt replied:
Wonderful, thanks. Fixed.

vitillo commented:
Tests are failing apparently.

vitillo commented:
@harterrt ^

harterrt commented:
@vitillo: Dag-gone, looks like all of the Cross Sectional tests passed but some other test issued a request to this URL[0] got a 500 response. I requested a re-count, which passed.

[0] https://hg.mozilla.org/releases/mozilla-beta/raw-file/tip/toolkit/components/telemetry/Scalars.yaml

vitillo commented:
Reviewed 1 of 3 files at r2.
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 10 at r2](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KTuNH2NtOLWHHrSJUox:-KTuNH2OMvTZB9VwkQfR:b-3zmkjo) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/f3d47bb63c22b7406eda5dce71b41088b8902897/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L10)):*
> ```Scala
> import com.mozilla.telemetry.utils.aggregation
> 
> case class ActiveAddons (
> ```

nit: I think this class should be named `ActiveAddon` as it represents a single add-on.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 53 at r2](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KTuN_U87vjs5KgEihGb:-KTuN_U9dpxDJPAxReZ-:bbokupe) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/f3d47bb63c22b7406eda5dce71b41088b8902897/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L53)):*
> ```Scala
>   }
> 
>   def addonCount() = {
> ```

Methods should have the return type listed even if it can be inferred (it's a convention). I noticed return types being missing for other methods as well; could you please add the types in a new commit?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122)*
<!-- Sent from Reviewable.io -->

vitillo commented:
Review status: 0 of 3 files reviewed at latest revision, 4 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 61 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KTuPEMF1PIH-JoF8iSc:-KTuPEMF1PIH-JoF8iSd:bjfi201) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/7256d586c48ae21852e8255b5c1cc4c2610a945e/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L61)):*
> ```Scala
>     this(
>         client_id = base.client_id
>       , normalized_channel = base.normalized_channel
> ```

Could you please move the commas up to reflect the convention used in the rest of the project?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122)*
<!-- Sent from Reviewable.io -->

harterrt commented:
Review status: 0 of 3 files reviewed at latest revision, 5 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 61 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KTuPEMF1PIH-JoF8iSc:-KTv4glBTUujDx9clF76:b-kwcpni) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/7256d586c48ae21852e8255b5c1cc4c2610a945e/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L61)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> Could you please move the commas up to reflect the convention used in the rest of the project?
</details>
To be clear, we want to revert the comma move to its previous state. I'll fix in the morning.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 23 at r2](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KTv4FPyTibIiiePAYZ0:-KTv4FPyTibIiiePAYZ1:b-2t268w) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/f3d47bb63c22b7406eda5dce71b41088b8902897/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L23)):*
> ```Scala
>   , val install_day: Option[Long]
>   , val update_day: Option[Long]
>   , val signed_state: Option[Long]
> ```

Need to add is_system to this class

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122)*
<!-- Sent from Reviewable.io -->

harterrt commented:
Review status: 0 of 3 files reviewed at latest revision, 5 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 61 at r1](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KTuPEMF1PIH-JoF8iSc:-KTyWhgzVQzihyoTs_UC:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/7256d586c48ae21852e8255b5c1cc4c2610a945e/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L61)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary>
> To be clear, we want to revert the comma move to its previous state. I'll fix in the morning.
</details>
Done.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 10 at r2](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KTuNH2NtOLWHHrSJUox:-KTyPaLuX-cQAYaZ6kTC:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/f3d47bb63c22b7406eda5dce71b41088b8902897/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L10)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> nit: I think this class should be named `ActiveAddon` as it represents a single add-on.
</details>
Done.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 23 at r2](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KTv4FPyTibIiiePAYZ0:-KTyXGEDBL3du3feG0DA:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/f3d47bb63c22b7406eda5dce71b41088b8902897/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L23)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary>
> Need to add is_system to this class
</details>
Done.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 53 at r2](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KTuN_U87vjs5KgEihGb:-KTyXJghsbIZERpLcZCK:b-g18a6w) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/f3d47bb63c22b7406eda5dce71b41088b8902897/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L53)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> Methods should have the return type listed even if it can be inferred (it's a convention). I noticed return types being missing for other methods as well; could you please add the types in a new commit?
</details>
Done. Will follow up with a clean-up PR soon.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122)*
<!-- Sent from Reviewable.io -->

vitillo commented:
Reviewed 3 of 3 files at r4.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 114 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KU2N5VZMMKY26vUEtJZ:-KU2N5V_RND_VKuEJAgv:b-ebqg9x) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d5218e18b716473cc4b17338824ddc6c2a622c6e/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L114)):*
> ```Scala
>       architecture_mode = base.weightedMode(base.architecture).getOrElse(None),
>       ffLocale_mode = base.weightedMode(base.locale).getOrElse(None),
>       addon_count_foreign_avg = base.foreignAddonCount.map(aggregation.mean _).getOrElse(None),
> ```

Why is mean preferred over a weighted mean?

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 115 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KU2LEGKSjupdgtbuGhF:-KU2LEGLmp1W1kV4JBzB:b-5ligav) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d5218e18b716473cc4b17338824ddc6c2a622c6e/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L115)):*
> ```Scala
>       ffLocale_mode = base.weightedMode(base.locale).getOrElse(None),
>       addon_count_foreign_avg = base.foreignAddonCount.map(aggregation.mean _).getOrElse(None),
>       addon_count_foreign_cfgs = base.foreignAddons.map(_.distinct.length),
> ```

Given `a: Map[String, ActiveAddon]` and  `b: Map[String, ActiveAddon]` with identical contents, are you sure that in all cases `a.values == b.values`? I.e. are values of a `Map` ordered?

---

*[src/test/scala/com/mozilla/telemetry/CrossSectionalViewTest.scala, line 50 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KU2Nji5PoPRzPK3_69f:-KU2Nji5PoPRzPK3_69g:bz3tut4) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d5218e18b716473cc4b17338824ddc6c2a622c6e/src/test/scala/com/mozilla/telemetry/CrossSectionalViewTest.scala#L50)):*
> ```Scala
>   }
> 
>   def getExampleLongitudinal(
> ```

This formatting looks confusing.

---

*[src/test/scala/com/mozilla/telemetry/CrossSectionalViewTest.scala, line 149 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KU2ODe_qIogNiZdKyt1:-KU2ODe_qIogNiZdKyt2:b-ekcrar) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d5218e18b716473cc4b17338824ddc6c2a622c6e/src/test/scala/com/mozilla/telemetry/CrossSectionalViewTest.scala#L149)):*
> ```Scala
>       addon_count_mode = Some(2)
>     )
>     println(actual)
> ```

Please remove prints.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122)*
<!-- Sent from Reviewable.io -->

harterrt commented:
Review status: 0 of 4 files reviewed at latest revision, 6 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 114 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KU2N5VZMMKY26vUEtJZ:-KUJtWeL9BIAurwSbxSs:b-zid30o) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d5218e18b716473cc4b17338824ddc6c2a622c6e/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L114)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> Why is mean preferred over a weighted mean?
</details>
Good catch. I fixed this in a future commit, but it's better to include it here. Fixed.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 115 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KU2LEGKSjupdgtbuGhF:-KUJxchUJJhQEIi7sFam:b4rpwe8) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d5218e18b716473cc4b17338824ddc6c2a622c6e/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L115)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> Given `a: Map[String, ActiveAddon]` and  `b: Map[String, ActiveAddon]` with identical contents, are you sure that in all cases `a.values == b.values`? I.e. are values of a `Map` ordered?
</details>
Good point, this implementation would double count configurations with the same foreign addons with different orderings. Only comparing values doesn't make sense, so I've updated the tests and fixed the helper function above.

Note, the [prototype implementation](https://github.com/mozilla/bcolloran_spark/blob/master/perClientDataPrep/cross-sectional%20data%20table%20%282016-07-14%29%20-%20deduped%20clientIds.ipynb) actually dedupes based on foreignAddonCount, so this is still slightly more sensitive than the prototype, but I think it makes more sense. 

---

*[src/test/scala/com/mozilla/telemetry/CrossSectionalViewTest.scala, line 50 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KU2Nji5PoPRzPK3_69f:-KUK1eKmkeyaqYAVCCaM:b-u1azws) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d5218e18b716473cc4b17338824ddc6c2a622c6e/src/test/scala/com/mozilla/telemetry/CrossSectionalViewTest.scala#L50)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> This formatting looks confusing.
</details>
I split the next line to match the style guide. Let me know if you meant something else. 

---

*[src/test/scala/com/mozilla/telemetry/CrossSectionalViewTest.scala, line 149 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KU2ODe_qIogNiZdKyt1:-KUJu5WPAF_Iu5yAIGMV:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d5218e18b716473cc4b17338824ddc6c2a622c6e/src/test/scala/com/mozilla/telemetry/CrossSectionalViewTest.scala#L149)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> Please remove prints.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122)*
<!-- Sent from Reviewable.io -->

harterrt commented:
@vitillo, RFAL

harterrt commented:
Review status: 0 of 4 files reviewed at latest revision, 6 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 115 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KU2LEGKSjupdgtbuGhF:-KUMkfie9wbHQRzUoP94:bdngpxm) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d5218e18b716473cc4b17338824ddc6c2a622c6e/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L115)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary>
> Good point, this implementation would double count configurations with the same foreign addons with different orderings. Only comparing values doesn't make sense, so I've updated the tests and fixed the helper function above.
> 
> Note, the [prototype implementation](https://github.com/mozilla/bcolloran_spark/blob/master/perClientDataPrep/cross-sectional%20data%20table%20%282016-07-14%29%20-%20deduped%20clientIds.ipynb) actually dedupes based on foreignAddonCount, so this is still slightly more sensitive than the prototype, but I think it makes more sense. 
</details>
Actually, I'm researching how we generate the map keys now. I'm not sure they are actually meaningful. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122)*
<!-- Sent from Reviewable.io -->

harterrt commented:
Review status: 0 of 4 files reviewed at latest revision, 5 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 115 at r4](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KU2LEGKSjupdgtbuGhF:-KUMm--xYQb6IFplml99:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/d5218e18b716473cc4b17338824ddc6c2a622c6e/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L115)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary>
> Actually, I'm researching how we generate the map keys now. I'm not sure they are actually meaningful. 
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122)*
<!-- Sent from Reviewable.io -->

vitillo commented:
Reviewed 4 of 4 files at r5.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/utils/Aggregation.scala, line 9 at r5](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KUMmbXhCEoU39Gj9Ttp:-KUMmbXhCEoU39Gj9Ttq:b-2t7sx) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/b6699815c2b5f101aa047b8eeae872642de5fb5b/src/main/scala/com/mozilla/telemetry/utils/Aggregation.scala#L9)):*
> ```Scala
>     val sum_prod = clean_pairs.foldLeft(0.0)((acc, pair) => acc + (pair._1 * pair._2))
> 
>     if ((ttl_weight <= 0) || (values.size != weights.size)) {
> ```

It might make sense to move the second condition up to avoid doing work that isn't needed.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 197 at r5](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KUMm1UZLMnwWgckOjED:-KUMm1UZLMnwWgckOjEE:b-h89c3n) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/b6699815c2b5f101aa047b8eeae872642de5fb5b/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L197)):*
> ```Scala
>     }
> 
>     val ex = output.cache.take(2)
> ```

Why do you need to cache here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122)*
<!-- Sent from Reviewable.io -->

harterrt commented:
@vitillo, RFAL

---

Review status: 2 of 4 files reviewed at latest revision, 4 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/utils/Aggregation.scala, line 9 at r5](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KUMmbXhCEoU39Gj9Ttp:-KUORNmk6VkZt_yxlLJJ:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/b6699815c2b5f101aa047b8eeae872642de5fb5b/src/main/scala/com/mozilla/telemetry/utils/Aggregation.scala#L9)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> It might make sense to move the second condition up to avoid doing work that isn't needed.
</details>
Done.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 18 at r2](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KTGO2KHfGmOQySIlX-1-r2-18:-KUOSWdv-AAi_2ZPhWbw:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/f3d47bb63c22b7406eda5dce71b41088b8902897/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L18)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary>
> Wonderful, thanks. Fixed.
</details>
Done.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 197 at r5](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KUMm1UZLMnwWgckOjED:-KUORSazjqgVFtryuyG-:b9jhcjv) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/b6699815c2b5f101aa047b8eeae872642de5fb5b/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L197)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> Why do you need to cache here?
</details>
Spark was only calculating the first row when doing a dry run. I discovered some errors in my code later than I would have liked because of this. As we've discussed, until we have [the fuzzer](https://bugzilla.mozilla.org/show_bug.cgi?id=1310324) the dry run is the only way to catch a certain class of errors.

I've moved the cache so it is only executed in a dry-run.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-:-KUOTC04R13NK-AbCaes:b-t3deqp)*
<!-- Sent from Reviewable.io -->

harterrt commented:
Review status: 2 of 4 files reviewed at latest revision, 4 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/utils/Aggregation.scala, line 9 at r5](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KUMmbXhCEoU39Gj9Ttp:-KURriFjbjgrBppuiVrP:b6a1b70) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/b6699815c2b5f101aa047b8eeae872642de5fb5b/src/main/scala/com/mozilla/telemetry/utils/Aggregation.scala#L9)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary>
> Done.
</details>
After sleeping on this, I think you may have wanted me to make the sum_prod calculation dependant on the ttl_weight being > 0. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122)*
<!-- Sent from Reviewable.io -->

vitillo commented:
Reviewed 2 of 2 files at r6.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 197 at r5](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KUMm1UZLMnwWgckOjED:-KURyiV1NkbYbe1sfyad:b-ahff59) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/b6699815c2b5f101aa047b8eeae872642de5fb5b/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L197)):*
<details><summary><i>Previously, harterrt (Ryan Harter) wrote…</i></summary>
> Spark was only calculating the first row when doing a dry run. I discovered some errors in my code later than I would have liked because of this. As we've discussed, until we have [the fuzzer](https://bugzilla.mozilla.org/show_bug.cgi?id=1310324) the dry run is the only way to catch a certain class of errors.
> 
> I've moved the cache so it is only executed in a dry-run.
</details>
I see, you could use `.count()` to force execution without stressing memory though. Either way you might want to add a comment about it for future reference.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122)*
<!-- Sent from Reviewable.io -->

harterrt commented:
@vitillo 

---

Review status: 3 of 4 files reviewed at latest revision, 3 unresolved discussions.

---

*[src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala, line 197 at r5](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-KUMm1UZLMnwWgckOjED:-KUSGa1K3BjpgNSPG4_U:b-896fix) ([raw file](https://github.com/mozilla/telemetry-batch-view/blob/b6699815c2b5f101aa047b8eeae872642de5fb5b/src/main/scala/com/mozilla/telemetry/views/CrossSectionalView.scala#L197)):*
<details><summary><i>Previously, vitillo (Roberto Agostino Vitillo) wrote…</i></summary>
> I see, you could use `.count()` to force execution without stressing memory though. Either way you might want to add a comment about it for future reference.
</details>
Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-:-KUSGc0Ryo0hY9VBBrbJ:bgx00no)*
<!-- Sent from Reviewable.io -->

vitillo commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/mozilla/telemetry-batch-view/122#-:-KUSLIMX753W6l8xYlDe:bnfp4nl)*
<!-- Sent from Reviewable.io -->
